src/libponyrt/sched/scheduler.c

```diff
@@ -491,3 +501,69 @@ PONY_API pony_ctx_t* pony_ctx()
   pony_assert(this_scheduler != NULL);
   return &this_scheduler->ctx;
 }
+
+void ponyint_sched_mute(pony_ctx_t* ctx, pony_actor_t* sender, pony_actor_t* recv)
+{
+  scheduler_t* sched = ctx->scheduler;
+  size_t index = HASHMAP_UNKNOWN;
+  muteref_t key;
+  key.key = recv;
+
+  muteref_t* mref = ponyint_mutemap_get(&sched->mute_mapping, &key, &index);
+  if(mref == NULL)
+  {
+    mref = ponyint_muteref_alloc(recv);
+    ponyint_mutemap_putindex(&sched->mute_mapping, mref, index);
+  }
+
+  size_t index2 = HASHMAP_UNKNOWN;
+  pony_actor_t* r = ponyint_muteset_get(&mref->value, sender, &index2);
+  if(r == NULL)
+  {
+    ponyint_muteset_putindex(&mref->value, sender, index2);
+    sender->muted += 1;
+  }
+}
+
+void ponyint_sched_unmute(pony_ctx_t* ctx, pony_actor_t* actor, bool inform)
+{
+  // this needs a better name. its not unmuting actor.
+  // its unmuting anything that got muted because of this actor
+  scheduler_t* sched = ctx->scheduler;
+  size_t index;
+  muteref_t key;
+  key.key = actor;
+
+  if(inform)
+  {
+    for(uint32_t i = 0; i < scheduler_count; i++)
+    {
+      if(&scheduler[i] != sched)
+        send_msg(i, SCHED_UNMUTE_ACTOR, (intptr_t)actor);
```

Comment:
Now that all schedulers can broadcast, `send_msg_single` is unsafe. Calls to `send_msg_single` should be replaced by calls to `send_msg`.

Reply:
does that mean that we should remove `send_msg_single` entirely? i believe that is the implication but I want to verify.

Reply:
I only find one instance in `send_msg_all` in scheduler.c

updating.

Reply:
Yes, this is what I meant.

Reply:
Its gone!